modules/light-clients/08-wasm/keeper/msg_server_test.go

```diff
@@ -79,20 +88,176 @@ func (suite *KeeperTestSuite) TestMsgStoreCode() {
+				}
+			},
+			nil,
+		},
+		{
+			"success: update state",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, newCodeHash, []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					// the code hash written in here will be overwritten
+					expClientState = types.NewClientState([]byte{1}, []byte(ibctesting.InvalidID), clienttypes.NewHeight(2000, 2))
+					store.Set(host.ClientStateKey(), clienttypes.MustMarshalClientState(suite.chainA.App.AppCodec(), expClientState))
+
+					data, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return &wasmvmtypes.Response{Data: data}, wasmtesting.DefaultGasUsed, nil
+				}
+			},
+			nil,
+		},
+		{
+			"failure: same code hash",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, oldCodeHash[:], []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					panic("unreachable")
+				}
+			},
+			types.ErrWasmCodeExists,
+		},
+		{
+			"failure: unauthorized signer",
+			func() {
+				msg = types.NewMsgMigrateContract(suite.chainA.SenderAccount.GetAddress().String(), defaultWasmClientID, newCodeHash, []byte("{}"))
+			},
+			ibcerrors.ErrUnauthorized,
+		},
+		{
+			"failure: invalid wasm code hash",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, []byte(ibctesting.InvalidID), []byte("{}"))
+			},
+			types.ErrWasmCodeHashNotFound,
+		},
+		{
+			"failure: invalid client id",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, ibctesting.InvalidID, newCodeHash, []byte("{}"))
+			},
+			clienttypes.ErrClientTypeNotFound,
+		},
+		{
+			"failure: contract returns error",
+			func() {
+				msg = types.NewMsgMigrateContract(govAcc, defaultWasmClientID, newCodeHash, []byte("{}"))
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					return nil, wasmtesting.DefaultGasUsed, wasmtesting.ErrMockContract
+				}
+			},
+			types.ErrWasmContractCallFailed,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupWasmWithMockVM()
+
+			newCodeHash = storeWasmCode(suite, newByteCode)
+
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
+
+			oldClientState, ok := endpoint.GetClientState().(*types.ClientState)
+			suite.Require().True(ok)
+
+			expClientState = oldClientState
```

Comment:
Since `oldClientState` is a pointer and you assign it to `expClientState`, then when modifying `CodeHash` below, you modify it in `oldClientState`. It's not a big deal, because `oldClientState` is not used, but if in the future it becomes used, it could cause some head scratching for a different developer. So why not something like this?

```golang
expClientState := func() *types.ClientState {
  cs, ok := endpoint.GetClientState().(*types.ClientState)
  suite.Require().True(ok)
  cs.CodeHash = newCodeHash
  return cs
}()
```

Reply:
`oldClientState` was unused. So I removed it